### ISSUE TITLE
Pytanie o ważność tokena

### ISSUE THREAD

**Issue by GregTora** [allegro/allegro-api#10491]
Dzień dobry,

mam dwa pytania:

1) Proszę o informację, czy po upływie 3 miesięcy "refresh token" sam przedłuży się na kolejne 3 miesiące? Czy trzeba wykonać jakąś akcję, aby został przedłużony?

![obraz](https://github.com/user-attachments/assets/827d90fa-998b-49ef-babd-69bd2a396e28)

2) Proszę o informację, czym jest aktywna sesja? To allegro uruchomione na jednej przeglądarce? Jak np. jeden użytkownik jest zalogowany na dwóch przeglądarkach to jest to liczone jako dwie sesje? I ilość otwartych zakładek w jednej przeglądarce ma znaczenie (no jak allegro jest otwarte na 25 zakładkach to wówczas limit jest przekroczony)?

![obraz](https://github.com/user-attachments/assets/25429447-7dd2-4f38-8936-8572fd193373)

**Comment by AureliuszBrussy** [comment]
1. Refresh token sam się nie przedłuża. Masz 3 miesiace na jego wykorzystanie w celu wygenerowania nowej pary access i refresh token. Więcej o tym, jako korzystać z refresh tokena przeczytasz w [naszym poradniku](https://developer.allegro.pl/tutorials/uwierzytelnianie-i-autoryzacja-zlq9e75GdIR#przedluzenie-waznosci-tokena).
2. Tak, zalogowanie się w 2 różnych przeglądarkach policzone zostanie jak 2 osobne sesje. Ilość otwarych zakładek nie ma znaczenia i nie wpływa na liczbę sesji. Limit sesji wynosi 20 jednocześnie. Jeżeli ponownie się zalogujesz (czy to przez WWW czy API) i powstanie 21 sesja, to sesja nr 1 straci swoją ważność. Nie bierzemy tu pod uwagę każde wywołanie zasobu, a generowaną parę access + refresh token.

**Comment by GregTora** [comment]
Mamy obecnie 5 aplikacji - trzy są z typem "DEVICE" i dwa z typem "WEB".

Proszę o informację, czy używanie aplikacji z typem "DEVICE" również się wlicza w limit zapytań/limit sesji?

**Comment by AureliuszBrussy** [comment]
Tak, również wlicza się w limity.

**Comment by AureliuszBrussy** [comment]
To zależy ile access tokenów zostało wygenrowanych równocześnie dla danego konta. Niestety, ale nie mamy funkcji, która pozowaliłaby to sprawdzić.

**Comment by GregTora** [comment]
Czyli wystarczy jak się dowiem ile access tokenów zostało wygenerowanych dla każdej aplikacji? Teoretycznie działa to tak, że jak dwie aplikacje mają wygenerowany acces token i na jeden aplikacji są 3 urządzenia, a na drugiej aplikacji 2 urządzenia to ilość się sumuje i łącznie jest 5 urządzeń?

**Comment by AureliuszBrussy** [comment]
W teorii to wystarczy, zależy to jednak od działania danej aplikacji.

**Comment by GregTora** [comment]
Poniżej screen z Allegro z informacją, że liczba zaufanych urządzeń to "16". Proszę o informację, czy jest to jednoznaczne z tym, że jest "zajętych/aktywnych" 16 sesji? Czy np jak na jednych z tych "16" urządzeń jest zalogowanie na dwóch przeglądarkach to wówczas jest policzone w sumie jako "17" sesji?

![obraz](https://github.com/user-attachments/assets/f0174cb3-9f01-4054-810c-9061812926a4)

**Comment by AureliuszBrussy** [comment]
Nie, to nie jest to samo. "Zaufane urządzenia" dotyczą tylko dwustopniowego logowania.

**Comment by GregTora** [comment]
A można w jakiś sposób w Allegro sprawdzić, ile na daną chwilę jest aktywnych sesji?

**Comment by AureliuszBrussy** [comment]
Nie, tak jak wspomniałem, nie jest to możliwe.

**Comment by GregTora** [comment]
ok, dziękuję za odpowiedź.

**Comment by GregTora** [comment]
Czy jak aplikacja API REST w Allegro jest utworzona, a nie jest używana to czy ona wlicza się do limitu sesji?

**Comment by AureliuszBrussy** [comment]
Nie, liczą się tylko aktualnie aktywne access tokeny. Jeżeli aplikacja ich nie generuje, to nie ma to znaczenia.
